HierarchicalDataConverter/Catalyst.HierarchicalDataConverter.AutomatedTests/GrandparentParentChildTest.cs

```diff
@@ -0,0 +1,441 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Catalyst.HierarchicalDataConverter.AutomatedTests
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Catalyst.DataProcessing.Shared.Models.Enums;
+    using Catalyst.DataProcessing.Shared.Models.Metadata;
+    using Catalyst.DataProcessing.Shared.Utilities.Client;
+
+    using DataConverter;
+
+    using Fabric.Databus.Config;
+
+    using Moq;
+
+    [TestClass]
+    public class GrandparentParentChildTest
+    {
+        [TestMethod]
+        public void TestMethod1()
+        {
+            Binding[] bindings = new Binding[3]
+                                     {
+                                         this.GetNestedBindingLevel0Source(),
+                                         this.GetNestedBindingLevel1Source(),
+                                         this.GetNestedBindingLevel2Source()
+                                     };
+            Field[] entity0Fields = this.GetEntity0Fields();
+            Field[] entity1Fields = this.GetEntity1Fields();
+            Field[] entity2Fields = this.GetEntity2Fields();
```

Comment:
if you're planning on having other tests using these same shared methods, you might want to consider putting them in a TestInitialize method. The mocks may be good to put there too.

Reply:
Good idea.  For now I'll leave them the way they are and refactor when needed.